Master add TIME_FORMAT config variable and fixes spelling errors in manual

Dloranger opened the pull request:
Adding a new config variable for setting up how the time is announced, gives option of using 12/24 hour formatting when announcing the time

sm0svx commented on src/doc/man/svxlink.conf.5:
```diff
@@ -1311,18 +1314,18 @@ The drawback is that the Ddr signal level is not completely comparable to the
 range. Set SIGLEV_DET=DDR to activate the Ddr signal level detector.
 .
-.SS Wideband Receiver Section
+.SS wide-band Receiver Section
```
Should be uppercased 'Wide-band"

sm0svx commented on src/doc/man/svxlink.conf.5:
```diff
@@ -1598,16 +1603,16 @@ much CPU time it should spend on doing a good job. Set it as high as possible
-bitrate when needed and decrease it when the quality can be assured with a
-lower bitrate. The target average bitrate is the one set by OPUS_ENC_BITRATE.
+bit-rate when needed and decrease it when the quality can be assured with a
+lower bit-rate. The target average bit-rate is the one set by OPUS_ENC_bit-rate.
```
Undo OPUS_ENC_BITRATE replace

sm0svx commented on src/doc/man/svxlink.conf.5:
```diff
@@ -1836,29 +1841,29 @@ challenge-response procedure will be used for authentication.
-Speex encoder setting. Specify the bitrate to use. Speex will snap to the
-nearest lower possible bitrate. Possible values range from 2150 to 24600 bps.
-You should probably not specify quality at the same time as bitrate. Not sure
+.B SPEEX_ENC_bit-rate
```
Undo SPEEX_ENC_BITRATE replace

sm0svx commented on src/doc/man/svxlink.conf.5:
```diff
@@ -1901,22 +1906,22 @@ much CPU time it should spend on doing a good job. Set it as high as possible
 .TP
-.B OPUS_ENC_BITRATE
-Opus encoder setting. This is the bitrate that the encoder will encode for.
+.B OPUS_ENC_bit-rate
```
Undo OPUS_ENC_BITRATE replace

sm0svx commented on src/svxlink/svxlink/locale.tcl:
```diff
@@ -129,37 +130,58 @@ proc playNumber {number} {
 # Say the time specified by function arguments "hour" and "minute".
 #
 proc playTime {hour minute} {
+  global time_format
```
You can implement this without modifying the C++ code since any configuration variable is available in the TCL code as CFG_varname. So if you specify TIME_FORMAT in the config you can access it as CFG_TIME_FORMAT in the TCL code. E.g.:
```
variable Logic::CFG_TIME_FORMAT
puts "### TIME_FORMAT=$CFG_TIME_FORMAT"
```
This also make this change tied to the English locale.tcl rather than "forcing" it on all langpacks.

Dloranger replied:
I fixed up the documentation issues, which also resolved the conflict.
~~~~~~~~
I do not understand how this change is "forcing" anything on all language packs,  the default behavior is maintained (12 hour is the default value). If the config variable is not the acceptable option of "24", it plays the 12 hour behavior, same as before.

If the other language packs don't bring in the new variable & code into their locale.tcl, wouldn't this new change be ignored anyways?

sm0svx replied:
Yes, but there is no reason to modify the C++ code in any case. What I meant with forcing (maybe the wrong word) was just that some languages may not have the concept of 12/24 hour format so it's language pack specific. Then it may be confusing if people think that they can use the TIME_FORMAT config variable for any langpack.

A little caveat is that the option still have to be documented somewhere and since there is no concept for documenting a langpack at the moment the only place is the svxlink.conf manual page. I guess a note about "may not work for all langpacks" will have to do for now.

Dloranger commented:
@sm0svx 

Please have a look again at this one.  I am not sure why it is complaining about the manual.  If you can understand, please let me know what I am doing wrong that is causing these conflicts when I am adding content.

sm0svx commented:
One problem may be that your git fork is not derived directly from my repository but rather from another fork. Just a guess though.

I'll see if I can figure it out tomorrow.

sm0svx commented:
Rebased, squashed and merged this PR to master. Also added the 'hours' sound-clip to the en_US-heather language pack.

I think you would get cleaner pull requests if you recreate your fork directly from sm0svx/svxlink. It's also important to always merge your branch with the master branch from sm0svx/svxlink. That should solve any remaining conflicts.

Dloranger commented:
Thanks, understood on the master branch, but I am developing in the non
sm0svx fork primarily and working to push as much up to you as I can to
help bring the features in that make sense for the trunk.

It appears I cannot fork from you directly without wiping out my own fork
where I am also developing more code that contains a lot of stuff that
doesn't make sense for your trunk like LCD integrations, driver
customizations, etc.

On Thu, Mar 8, 2018, 1:32 PM Tobias Blomberg <notifications@github.com>
wrote:

> Rebased, squashed and merged this PR to master. Also added the 'hours'
> sound-clip to the en_US-heather language pack.
>
> I think you would get cleaner pull requests if you recreate your fork
> directly from sm0svx/svxlink. It's also important to always merge your
> branch with the master branch from sm0svx/svxlink. That should solve any
> remaining conflicts.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/sm0svx/svxlink/pull/357#issuecomment-371631308>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AJ6NjRvQOpeV1Ouf81IcqVIQGNVmR7cLks5tcaNRgaJpZM4R4vSd>
> .
>

sm0svx commented:
Ok, I understand. You should be able to provide clean PRs anyway but you need to take care when creating branches. Assuming you have a Git remote setup for my repo called 'sm0svx' I'd do something like this (not tested so you may need to tweak it).
```
git fetch sm0svx
git checkout -b my-new-branch sm0svx/master
# Work work work
git commit -a
git merge sm0svx/master
git push origin
# Create pull request
```
If you already have a branch that is not based on sm0svx/master that you want to create a PR for, you can rebase that branch onto the tip of sm0svx/master before creating a PR.
```
git branch my-branch-backup
git rebase sm0svx/master
```
The first row is just for creating a backup of the original branch pointer in case you mess up the rebase completely. Note that a rebase will change history so if you have a previously pushed copy of that branch you will have to force push it after the rebase. If there are other people using that branch they will get into trouble so rebasing an existing pushed branch is normally not recommended, but if you know what you are doing it's a powerful tool.

Dloranger commented:
I will have a look at this when I do future ones.  I am working from the Github desktop so I will need to learn a bit about working in the console version of git.

Is this rebase safe to do with my existing config_id's branch/PR?  I am getting the same conflict for the manual only.
